File: server/dotnet/Controllers/Render.cs
using System.Text;
using Microsoft.AspNetCore.Mvc;

public class RenderController : Controller
{
    private readonly TemplatePathResolver _templatePathResolver;

    public RenderController(TemplatePathResolver templatePathResolver)
    {
        _templatePathResolver = templatePathResolver;
    }

    [HttpGet("/")]
    public async Task<IActionResult> SdkRender()
    {
        var isFlexibleIntegration = false;
        var queryParams = HttpContext.Request.Query;

        foreach (var param in queryParams)
        {
            if (param.Key == "flexible") {
                isFlexibleIntegration = true;
            }
        }

        var locals = new Dictionary<string, string>
        {
            { "title", "Fastlane - Braintree SDK Integration" + (isFlexibleIntegration ? " (Flexible)" : "") },
            { "prerequisiteScripts", @"
                <script
                    src=""https://js.braintreegateway.com/web/3.104.0/js/client.min.js""
                    defer
                ></script>
                <script
                    src=""https://js.braintreegateway.com/web/3.104.0/js/data-collector.min.js""
                    defer
                ></script>
                <script
                    src=""https://js.braintreegateway.com/web/3.104.0/js/fastlane.min.js""
                    defer
                ></script>
            " },
            { "initScriptPath", isFlexibleIntegration ? "init-fastlane-flexible.js" : "init-fastlane.js" },
            { "stylesheetPath", "styles.css" }
        };

        var renderedHtml = await _templatePathResolver.RenderTemplateAsync(isFlexibleIntegration, locals);

        return Content(renderedHtml, "text/html", Encoding.UTF8);
    }
}

File: server/python/controllers/render.py
import os
import chevron


def braintree_sdk_render(args, templates_folder):
    is_flexible_integration = "flexible" in args

    locals = {
        "title": "Fastlane - Braintree SDK Integration"
        + (" (Flexible)" if is_flexible_integration else ""),
        "prerequisiteScripts": """
            <script
                src="https://js.braintreegateway.com/web/3.104.0/js/client.min.js"
                defer
            ></script>
            <script
                src="https://js.braintreegateway.com/web/3.104.0/js/data-collector.min.js"
                defer
            ></script>
            <script
                src="https://js.braintreegateway.com/web/3.104.0/js/fastlane.min.js"
                defer
            ></script>
        """,
        "initScriptPath": (
            "init-fastlane-flexible.js"
            if is_flexible_integration
            else "init-fastlane.js"
        ),
        "stylesheetPath": "styles.css",
    }

    template_name = (
        "checkout-flexible.html" if is_flexible_integration else "checkout.html"
    )
    template_path = os.path.join(templates_folder, template_name)

    with open(template_path, "r") as f:
        template = f.read()

    html = chevron.render(template, locals)
    return html
